### PR TITLE
io.h for VxWorks

### PR DESCRIPTION
**Pull request by kojo1**
Zendesk #3052
- io.h for VxWorks
- settings.h WOLFSSL_HAVE_MAX

### CODE REVIEW

File: wolfssl/io.h [hunk 1]
```diff
@@ -234,6 +234,9 @@
 #elif defined(FREERTOS_TCP)
     #define RECV_FUNCTION(a,b,c,d)  FreeRTOS_recv((Socket_t)(a),(void*)(b), (size_t)(c), (BaseType_t)(d))
     #define SEND_FUNCTION(a,b,c,d)  FreeRTOS_send((Socket_t)(a),(void*)(b), (size_t)(c), (BaseType_t)(d))
+#elif defined(WOLFSSL_VXWORKS)
```
Comment (cconlon) [inline wolfssl/io.h]: Why did VXWORKS need its own section here, instead of falling through to the #else condition?  It looks like it just uses send/recv like the else case does.

File: wolfssl/wolfcrypt/settings.h [hunk 2]
```diff
@@ -354,6 +356,7 @@
     #define SINGLE_THREADED
     #define NO_WOLFSSL_DIR
     #define WOLFSSL_HAVE_MIN
+    #define WOLFSSL_HAVE_MAX
```
Comment (cconlon) [inline wolfssl/wolfcrypt/settings.h]: This is inside the WOLFSSL_UTASKER define.  As this is not related to VxWorks, did you test this with the uTasker OS and compiler?

File: wolfssl/wolfcrypt/settings.h [hunk 3]
```diff
@@ -210,6 +210,7 @@
     #define TFM_TIMING_RESISTANT
     #define NEED_AES_TABLES
     #define WOLFSSL_HAVE_MIN
+    #define WOLFSSL_HAVE_MAX
```
Comment (cconlon) [inline wolfssl/wolfcrypt/settings.h]: This is inside the MICROCHIP_PIC32 define.  David has also changed this in his PR here:
https://github.com/wolfSSL/wolfssl/pull/357

File: wolfssl/io.h (hunk 1 above)
Comment (kojo1) [inline wolfssl/io.h]: If this fall into #else, 
#define HAVE_SOCKADDR part causes duplicated define of
typedef struct sockaddr         SOCKADDR;

They are defined in their header.

File: wolfssl/wolfcrypt/settings.h (hunk 3 above)
Comment (kojo1) [inline wolfssl/wolfcrypt/settings.h]: Thanks for the note. Will remove this from this PR.

File: wolfssl/wolfcrypt/settings.h (hunk 2 above)
Comment (kojo1) [inline wolfssl/wolfcrypt/settings.h]: Will remove this. I don't have the compiler.
